server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -165,6 +166,11 @@ public interface PluginServices {
          * A provider of utilities to observe and report parsing of documents
          */
         DocumentParsingProvider documentParsingProvider();
+
+        /**
+         * The task manager for the node
```

Comment:
Can you add a comment here that this should _only_ be used for tracking task removal? In the future we intend to make listener registration like this automatic so that we don't need to expose the full blown TaskManager, but for now we have to expose it here.

Reply:
Could we use the new DI mechanisms here, instead of adding the method and then updating it later?

Reply:
Listener support in DI will be such a big win!